Build error for Windows targeting CY2024

### Discussed in https://github.com/AcademySoftwareFoundation/OpenRV/discussions/411

<div type='discussions-op-text'>

<sup>Originally posted by **ronBublitz** March 13, 2024</sup>
Getting this error when building:
 "C:\OpenRV2.0\_build\main_executable.vcxproj" (default target) (1) ->
       "C:\OpenRV2.0\_build\src\bin\apps\rv\rv.vcxproj" (default target) (3) ->
       "C:\OpenRV2.0\_build\executables_with_plugins.vcxproj" (default target) (25) ->
       "C:\OpenRV2.0\_build\src\bin\imgtools\rvio\rvio.vcxproj" (default target) (120) ->
       "C:\OpenRV2.0\_build\src\lib\ip\OCIONodes\OCIONodes.vcxproj" (default target) (237) ->
       (ClCompile target) ->
         C:\OpenRV2.0\src\lib\ip\OCIONodes\OCIO1DLUT.cpp(24,15): error C2660: 'OpenColorIO_v2_3::Gp
       uShaderDesc::getTexture': function does not take 7 arguments [C:\OpenRV2.0\_build\src\lib\ip
       \OCIONodes\OCIONodes.vcxproj]

Using the repo as of March 13, 2024.
VS Studio 2022 Community

I was able to successfully build RV2.0 with the downloaded repo without any changes.  However, I wanted to test the changes when the CY2024 was specified (in /cmake/defaults/rv_options.cmake).  The error above is what was output after doing a clean build.


</div>

--- a/src/lib/ip/OCIONodes/OCIO1DLUT.cpp
+++ b/src/lib/ip/OCIONodes/OCIO1DLUT.cpp
@@ -21,8 +21,16 @@ namespace IPCore
     OCIO::GpuShaderDesc::TextureType textureType =
         OCIO::GpuShaderDesc::TEXTURE_RGB_CHANNEL;
     OCIO::Interpolation interpolation = OCIO::INTERP_BEST;
+#if defined( RV_VFX_CY2023 )
+    // Older (pre OCIOv2 2.3.x) getTexture() function signature
     shaderDesc->getTexture( idx, textureName, samplerName, width, height,
                             textureType, interpolation );
+#else
+    OCIO::GpuShaderDesc::TextureDimensions _textureDim = 
+        OCIO::GpuShaderDesc::TextureDimensions::TEXTURE_1D;
+    shaderDesc->getTexture( idx, textureName, samplerName, width, height,
+                            textureType, _textureDim, interpolation );
+#endif
     if( !samplerName || !*samplerName || width == 0 )
     {
       TWK_THROW_EXC_STREAM( "The OCIO 1D LUT texture data is corrupted" );

